data-reporting-service/post-processing-service/src/main/java/gov/cdc/etldatapipeline/postprocessingservice/service/PostProcessingService.java

```diff
@@ -0,0 +1,150 @@
+    }
+
+    @KafkaListener(topics = {"${spring.kafka.topic.investigation-confirmation}"})
+    public void postProcessInvestigationConfirmationMessage(@Header(KafkaHeaders.RECEIVED_KEY) String key,
+                                                            @Header(KafkaHeaders.RECEIVED_TOPIC) String topic) {
+        Long id = extractIdFromMessage(key, topic);
+        if(id != null) {
+            idCache.computeIfAbsent(topic, k -> new CopyOnWriteArrayList<>()).add(id);
+        }
+    }
+
+    @KafkaListener(topics = {"${spring.kafka.topic.investigation-notification}"})
+    public void postProcessInvestigationNotificationMessage(@Header(KafkaHeaders.RECEIVED_KEY) String key,
+                                                            @Header(KafkaHeaders.RECEIVED_TOPIC) String topic) {
+        Long id = extractIdFromMessage(key, topic);
+        if(id != null) {
+            idCache.computeIfAbsent(topic, k -> new CopyOnWriteArrayList<>()).add(id);
+        }
+    }
+
+    @KafkaListener(topics = {"${spring.kafka.topic.organization}"})
+    public void postProcessOrganizationMessage(@Header(KafkaHeaders.RECEIVED_KEY) String key,
+                                               @Header(KafkaHeaders.RECEIVED_TOPIC) String topic) {
+        Long id = extractIdFromMessage(key, topic);
+        if(id != null) {
+            idCache.computeIfAbsent(topic, k -> new CopyOnWriteArrayList<>()).add(id);
+        }
+    }
+
+    @KafkaListener(topics = {"${spring.kafka.topic.patient}"})
+    public void postProcessPatientMessage(@Header(KafkaHeaders.RECEIVED_KEY) String key,
+                                          @Header(KafkaHeaders.RECEIVED_TOPIC) String topic) {
+        Long id = extractIdFromMessage(key, topic);
+        if(id != null) {
+            idCache.computeIfAbsent(topic, k -> new CopyOnWriteArrayList<>()).add(id);
+        }
+    }
+
+    @KafkaListener(topics = {"${spring.kafka.topic.provider}"})
+    public void postProcessProviderMessage(@Header(KafkaHeaders.RECEIVED_KEY) String key,
+                                          @Header(KafkaHeaders.RECEIVED_TOPIC) String topic) {
+        Long id = extractIdFromMessage(key, topic);
+        if(id != null) {
+            idCache.computeIfAbsent(topic, k -> new CopyOnWriteArrayList<>()).add(id);
+        }
+    }
+
+    @Scheduled(fixedDelay = 5000)
+    protected void processCachedIds() {
+        for (Map.Entry<String, List<Long>> entry : idCache.entrySet()) {
+            if(!entry.getValue().isEmpty()) {
+                String keyTopic = entry.getKey();
+                List<Long> ids = entry.getValue();
+                idCache.put(keyTopic, new ArrayList<>());
+                if(keyTopic.contains("investigation")) {
+                    String idsString = String.join(",", ids.stream().map(String::valueOf).collect(Collectors.toList()));
+                    logger.info("Processing the ids from the topic {} and calling the stored proc for patient: {}", keyTopic, idsString);
```

Comment:
Small log text update to reflect investigation code block

Reply:
Investigation is still WIP